src/main/java/org/codehaus/mojo/flatten/ModelBuilderThreadSafetyWorkaround.java

```diff
@@ -83,7 +82,7 @@
     private SuperPomProvider superPomProvider;
 
     @Inject
-    private InheritanceAssembler inheritanceAssembler;
+    private DirectDependenciesInheritanceAssembler inheritanceAssembler;
```

Comment:
Maybe using a `@Named` annotation instead of the implementation class to choose which implementation is to be injected ?

Reply:
I get a NullPointer when using @Named instead of @Inject and using InheritanceAssembler

Cannot invoke "org.apache.maven.model.inheritance.InheritanceAssembler.assembleModelInheritance(org.apache.maven.model.Model, org.apache.maven.model.Model, org.apache.maven.model.building.ModelBuildingRequest, org.apache.maven.model.building.ModelProblemCollector)" because "this.inheritanceAssembler" is null

Reply:
It works with `@Component( role = InheritanceAssembler.class )` on the `DirectDependenciesInheritanceAssembler`

Reply:
In the `FlattenMojo` there is a direct usage of the implementation, yes.